packages/dashboard/src/components/iot-dashboard/iot-dashboard-wrapper.tsx

```diff
@@ -44,35 +47,116 @@ export class IotDashboardWrapper {
   /** Width and height of the cell, in pixels */
   @Prop() cellSize: number = DEFAULT_CELL_SIZE;
 
+  @Prop() addWidget: () => void;
+
   @State() dashboardLayout: DashboardConfiguration;
 
+  /** Selection gesture */
+  @State() start: Position | undefined;
+  @State() end: Position | undefined;
+  @State() finishedSelecting: boolean = false;
+  @State() previousPosition: Position | undefined;
+
+  /** List of ID's of the currently selected widgets. */
+  @State() selectedWidgetIds: string[] = [];
+
+  @State() currWidth: number;
+
+  @State() intermediateLayout: DashboardConfiguration | undefined;
+
+  /** The currently active gesture */
+  @State() activeGesture: 'move' | 'resize' | undefined;
+  /** If the active gesture is resize, this represents which anchor point the resize is being done relative to */
+  @State() activeResizeAnchor: Anchor | undefined;
+  /** The initial position of the cursor on the start of the resize gesture */
+  @State() resizeStartPosition: Position | undefined;
+
+  actualCellSize = () => {
+    const scale = this.stretchToFit ? this.currWidth / this.width : 1;
+    return scale * this.cellSize;
+  };
+
+  move(moveInput: MoveActionInput) {
+    this.store.dispatch(onMoveAction(moveInput));
+    this.previousPosition = moveInput.position;
+  }
+  moveWidgets(moveInput: MoveActionInput) {
+    this.dashboardLayout = getMovedDashboardConfiguration({
+      dashboardConfiguration: this.dashboardLayout,
+      position: moveInput.position,
+      previousPosition: moveInput.prevPosition,
+      selectedWidgetIds: moveInput.widgetIds,
+      cellSize: moveInput.cellSize,
+    });
+    this.onDashboardConfigurationChange(this.dashboardLayout);
+  }
+  midResize(resizeInput: ResizeActionInput) {
+    this.intermediateLayout = resize({
```

Comment:
is there a reason not to also do this in the reducer?

Reply:
when i did this through the reducer, it updated the dashboard configuration while it was resizing which made the widgets resize exponentially.